include/zephyr/drivers/rtc.h

```diff
@@ -0,0 +1,510 @@
+ */
+
+/**
+ * @typedef rtc_alarm_irq_callback
+ * @brief RTC alarm triggered IRQ callback
+ *
+ * @param dev Device instance invoking the handler
+ * @param id Alarm id
+ * @param user_data Optional user data passed with the alarm configuration
+ */
+typedef void (*rtc_alarm_irq_callback)(const struct device *dev, uint16_t id, void *user_data);
+
+/**
+ * @typedef rtc_time_set
+ * @brief API for setting RTC time
+ */
+typedef int (*rtc_api_time_set)(const struct device *dev, const struct tm *timeptr);
+
+/**
+ * @typedef rtc_time_get
+ * @brief API for getting RTC time
+ */
+typedef int (*rtc_api_time_get)(const struct device *dev, struct tm *timeptr);
+
+/**
+ * @typedef rtc_alarm_time_set
+ * @brief API for setting RTC alarm time
+ */
+typedef int (*rtc_api_alarm_time_set)(const struct device *dev, uint16_t id, uint16_t mask,
+				      const struct tm *timeptr);
+
+/**
+ * @typedef rtc_alarm_time_get
+ * @brief API for getting RTC alarm time
+ */
+typedef int (*rtc_api_alarm_time_get)(const struct device *dev, uint16_t id, uint16_t *mask,
+				      struct tm *timeptr);
+
+/**
+ * @typedef rtc_alarm_enable
+ * @brief API for enabling RTC alarm
+ */
+typedef int (*rtc_api_alarm_enable)(const struct device *dev, uint16_t id);
+
+/**
+ * @typedef rtc_alarm_disable
+ * @brief API for disabling RTC alarm
+ */
+typedef int (*rtc_api_alarm_disable)(const struct device *dev, uint16_t id);
+
+/**
+ * @typedef rtc_alarm_is_enabled
+ * @brief API for testing if RTC alarm is enabled
+ */
+typedef int (*rtc_api_alarm_is_enabled)(const struct device *dev, uint16_t id);
+
+/**
+ * @typedef rtc_alarm_is_pending
+ * @brief API for testing if RTC alarm is pending
+ */
+typedef int (*rtc_api_alarm_is_pending)(const struct device *dev, uint16_t id);
+
+/**
+ * @typedef rtc_alarm_clear_pending
+ * @brief API for clearing RTC alarm pending status
+ */
+typedef int (*rtc_api_alarm_clear_pending)(const struct device *dev, uint16_t id);
+
+/**
+ * @typedef rtc_alarm_irq_callback_set
+ * @brief API for setting alarm IRQ callback
+ */
+typedef int (*rtc_api_alarm_irq_callback_set)(const struct device *dev,
+					      rtc_alarm_irq_callback callback, void *user_data);
+
+/**
+ * @typedef rtc_alarm_irq_enable
+ * @brief API for enabling RTC alarm interrupt
+ */
+typedef int (*rtc_api_alarm_irq_enable)(const struct device *dev, uint16_t id);
+
+/**
+ * @typedef rtc_alarm_irq_disable
+ * @brief API for disabling RTC alarm interrupt
+ */
+typedef int (*rtc_api_alarm_irq_disable)(const struct device *dev, uint16_t id);
+
+/**
+ * @typedef rtc_calibration_set
+ * @brief API for setting RTC calibration
+ */
+typedef int (*rtc_api_calibration_set)(const struct device *dev, int32_t calibration);
+
+/**
+ * @typedef rtc_calibration_get
+ * @brief API for getting RTC calibration
+ */
+typedef int (*rtc_api_calibration_get)(const struct device *dev, int32_t *calibration);
+
+/**
+ * @brief RTC driver API
+ */
+__subsystem struct rtc_driver_api {
+	rtc_api_time_set time_set;
```

Comment:
@keith-zephyr points out `subject_verb_object` is usually preferred, so `rtc_api_set_time` instead here, and similarly elsewhere